src/ts/src/common/types/ui.ts

```diff
@@ -0,0 +1,16 @@
+import { FC } from "react";
+
+export interface ButtonInfo {
+  content?: string | JSX.Element;
+  type?: "primary" | "secondary";
+  link?: string | "cancel";
+  Button?: FC;
+}
+
+export interface ModalInfo {
+  header?: string | JSX.Element;
+  body?: string | JSX.Element;
+  buttons?: ButtonInfo[];
+  width?: number;
+  backgroundColor?: string;
+}
```

Comment:
This should be a `.d.ts` file, in line with the TypeScript convention used in the other files. If you want to limit the type to `React.FunctionComponent`, you can reference React types with the triple-slash import directive. https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html. Or, you can use the simpler `JSX.Element`, which encompasses a React FC.

Reply:
this was from a previous PR that @tihuan wrote, would it be ok if we patch it in a separate PR?

Reply:
Ah we actually don't need `d.ts`, since our code is written in Typescript instead of typing a non typescript library!

https://react-typescript-cheatsheet.netlify.app/docs/advanced/misc_concerns/#working-with-non-typescript-libraries-writing-your-own-indexdts

Reply:
Yeah that's the most common usage of `.d.ts` files, yeah. The reason I used them was because those files are "ambient" - the declarations there are available everywhere in the project without having to import them. If this works and it's the style you want to go with, that's fine. I would like to keep the way we declare global types consistent though, so if you want to change up things for this please change them for all the other type declarations.

Reply:
Ahh got it!

I try to steer clear from putting things in global namespace as much as possible to avoid global namespace collision and I find the import statements make things are more explicit too 😆 In theory we could also use Typescript `namespace` to manage collisions, but naming is hard and that's another layer of work we need to manage and could avoid for free by importing type definitions. For ambient or other commonly used entities, in the past projects we use `src/common/entities/*` to host them, so I'm thinking we could do the same thing here?

Keeping type definitions colocated with their code in the same file also has the added benefit of making it easier for engineers to update things and predictably find and use the definitions as well. So I find it easier to work with a codebase that way!

Love the idea of keeping type declaration consistent 👍 Will update other type definitions! Thank you 😄